src/components/Button/index.tsx

```diff
@@ -23,61 +23,61 @@ export interface BaseButtonProps extends BaseComponentProps {
   ariaLabel?: string;
 
   /**
-   * Button size. Defaults to regular.
+   * The color describing the button's intent. Defaults to primary.
    */
-  size?: ButtonSize;
+  color?: ButtonColor;
 
   /**
-   * The color describing the button's intent. Defaults to primary.
+   * Optional icon.
    */
-  color?: ButtonColor;
+  icon?: IconName;
 
   /**
-   * Click callback handler. Will provide the React synthetic event.
+   * Icon position.
    */
-  onClick?: (event: any) => void;
+  iconPosition?: ButtonIconPosition;
 
   /**
-   * Hover callback handler. Will provide the React synthetic event.
+   * Button size. Defaults to regular.
    */
-  onMouseEnter?: (event: any) => void;
+  size?: ButtonSize;
 
   /**
-   * Hover out callback handler. Will provide the React synthetic event.
+   * Focus lost callback handler. Will provide the React synthetic event.
    */
-  onMouseLeave?: (event: any) => void;
+  onBlur?: (event: any) => void;
 
   /**
-   * Focus callback handler. Will provide the React synthetic event.
+   * Click callback handler. Will provide the React synthetic event.
    */
-  onFocus?: (event: any) => void;
+  onClick?: (event: any) => void;
 
   /**
-   * Focus lost callback handler. Will provide the React synthetic event.
+   * Focus callback handler. Will provide the React synthetic event.
    */
-  onBlur?: (event: any) => void;
+  onFocus?: (event: any) => void;
 
   /**
-   * Optional icon.
+   * Hover callback handler. Will provide the React synthetic event.
    */
-  icon?: IconName;
+  onMouseEnter?: (event: any) => void;
 
   /**
-   * Icon position.
+   * Hover out callback handler. Will provide the React synthetic event.
    */
-  iconPosition?: ButtonIconPosition;
+  onMouseLeave?: (event: any) => void;
 }
 
 export interface RegularButtonProps extends BaseButtonProps {
   /**
-   * Whether this button should be disabled or not. Defaults to false.
+   * Disabled buttons cannot have a link href.
    */
-  disabled?: boolean;
+  href?: void;
```

Comment:
why is this void and not a string?

Reply:
We discussed in person. We think they're there to disambiguate the union.